tests/components/conversation/test_trigger.py

```diff
@@ -0,0 +1,55 @@
+"""Test sentence triggers."""
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components import conversation
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import (
+    intent,
+)
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture
+async def init_components(hass: HomeAssistant):
+    """Initialize relevant components with empty configs."""
+    assert await async_setup_component(hass, "homeassistant", {})
+    assert await async_setup_component(hass, "conversation", {})
+    assert await async_setup_component(hass, "intent", {})
+
+
+async def test_register_trigger(hass: HomeAssistant, init_components) -> None:
+    """Test registering/unregistering/matching a few trigger sentences."""
+    callback = MagicMock()
+
+    trigger_sentences = ["It's party time", "It is time to party"]
+    trigger_response = "Cowabunga!"
+    unregister = await conversation.async_register_trigger_sentences(
+        hass, trigger_sentences, callback, response=trigger_response
+    )
+
+    result = await conversation.async_converse(hass, "Not the trigger", None, Context())
```

Comment:
This is not testing the trigger.

Reply:
Use `async_initialize_triggers` from `helpers/trigger.py` to initialize a trigger.

Reply:
Moved this test into the default agent tests. I'll coordinate with @emontnemery for a proper trigger test.